wp-includes/class-wp-web-app-manifest.php

```diff
@@ -83,15 +83,51 @@ public function manifest_link_and_meta() {
 				?>
 				<meta name="apple-mobile-web-app-capable" content="yes">
 				<meta name="mobile-web-app-capable" content="yes">
+
 				<?php
 				$icons = isset( $manifest['icons'] ) ? $manifest['icons'] : array();
 				usort( $icons, array( $this, 'sort_icons_callback' ) );
 				$icon = array_shift( $icons );
-				?>
-				<?php if ( ! empty( $icon ) ) : ?>
-					<link rel="apple-touch-startup-image" href="<?php echo esc_url( $icon['src'] ); ?>">
-				<?php endif; ?>
-				<?php
+
+				$images = array();
+				if ( ! empty( $icon ) ) {
+					$images[] = array( 'href' => $icon['src'] );
+				}
+
+				/**
+				 * Filters splash screen images for Safari on iOS.
+				 *
+				 * @param array $images {
+				 *     Array of splash screen images and their attributes.
+				 *
+				 *     @type array ...$0 {
+				 *         Array of splash screen image attributes.
+				 *
+				 *         @type string $href  URL of splash screen image. Required.
+				 *         @type string $media Media query for when the splash screen image should be used.
+				 *     }
+				 * }
+				 */
+				$images = apply_filters( 'apple_touch_startup_image', $images );
+
+				foreach ( $images as $key => $image ) {
+					if ( ! is_array( $image ) ) {
+						continue;
+					}
+
+					$url = isset( $image['href'] ) ? esc_url( $image['href'], array( 'http', 'https' ) ) : '';
+					if ( ! $url ) {
+						continue;
+					}
+
+					$html = sprintf( 'rel="apple-touch-startup-image" href="%s"', $url );
```

Comment:
This seems safer to be more resilient against future changes which could accidentally make `$url` cause escaping issues.

```suggestion
					$html = sprintf( 'rel="apple-touch-startup-image" href="%s"', esc_attr( $url ) );
```

Reply:
I think whether this is more resilient is debatable (I would argue `esc_attr( $url )` looks confusing to most WordPress developers since all documentation says to use `esc_url()` for URLs), but I will rewrite this code to separate the URL used for validation and the URL used for display.